sieve_of_eratosthenes/sieveOfEratosthenes.c

```diff
@@ -0,0 +1,34 @@
+#include <stdio.h>
+#include <string.h>
+#include <math.h>     // for using square root 
+#include <stdbool.h>  // to include bool datatype 
+
+void sieveOfEratosthenes(int n) {
+	bool primes[n+1];
+	int i, sqrtOfn = sqrt(n)+1, j;
+	
+	memset(primes, true, sizeof(primes));
```

Comment:
> This works fine here because the size of the bool is 1 byte.

But it will be calculating `sizeof(bool*)` as `primes` is an array of `bool`. I still don't get that how it sets the entire array to `true`. Can someone explain.